awscli/autoprompt/widgets.py

```diff
@@ -195,3 +213,97 @@ def __init__(self):
 
     def __pt_container__(self):
         return self.container
+
+
+class DebugPanelWidget:
+    DIMENSIONS = {
+        'width': Dimension(max=40, preferred=20),
+    }
+
+    def __init__(self):
+        self._panel_prev_focus = None
+        self._dialog_prev_focus = None
+
+        self.dialog = self.create_save_file_dialog()
+
+        _kb = KeyBindings()
+        _kb.add(Keys.ControlS, filter=is_debug_mode, is_global=True)(
+            self._activate_dialog)
+        _kb.add(Keys.ControlBackslash, filter=is_debug_mode, is_global=True)(
+            self._switch_to_debug_panel)
+
+        self.float_container = FloatContainer(
+            Window(
+                content=BufferControl(
+                    buffer=Buffer(name='debug_buffer', read_only=True)
+                ),
+                wrap_lines=True,
+            ),
+            key_bindings=_kb,
+            floats=[]
+        )
+        self.container = ConditionalContainer(
+            Frame(
+                HSplit([
+                    self.float_container,
+                    HorizontalLine(),
+                    DebugToolbarView()
+                ]),
+                **self.DIMENSIONS,
+                title='Debug panel'
+            ),
+            filter=is_debug_mode
+        )
+
+    def _activate_dialog(self, event):
+        layout = event.app.layout
+        self._dialog_prev_focus = event.app.current_buffer
+        self.float_container.floats.append(Float(content=self.dialog))
+        layout.focus(self.dialog)
+
+    def _switch_to_debug_panel(self, event):
+        current_buffer = event.app.current_buffer
+        layout = event.app.layout
+        if current_buffer.name != 'debug_buffer':
+            self._panel_prev_focus = current_buffer
+            debug_buffer = layout.get_buffer_by_name('debug_buffer')
+            layout.focus(debug_buffer)
+        else:
+            layout.focus(self._panel_prev_focus)
+
+    def create_save_file_dialog(self):
+        default_filename = 'prompt_debug.log'
+
+        def cancel_handler():
+            app = get_app()
+            textfield.text = default_filename
+            app.layout.focus(textfield)
+            self.float_container.floats.pop()
+            app.layout.focus(self._dialog_prev_focus)
+
+        def ok_handler():
+            app = get_app()
+            buffer = app.layout.get_buffer_by_name('debug_buffer')
+            with open(textfield.text, 'w') as f:
```

Comment:
We should make sure to expand both the absolute path and user for this file provided. I ran into an error when I typed `~/debug.log` in the dialogue.